hazelcast/src/main/java/com/hazelcast/sql/impl/client/SqlClientService.java

```diff
@@ -282,6 +297,14 @@ private void handleExecuteResponse(SqlClientResult res, ClientMessage message) {
                     sqlError.getSuggestion()
             );
         } else {
+            if (response.partitionArgumentIndex != originalPartitionArgumentIndex) {
+                if (response.partitionArgumentIndex != -1) {
+                    partitionArgumentIndexCache.put(sqlText, response.partitionArgumentIndex);
```

Comment:
We can avoid adding anything to the case, if this client is a unisocket client (seee `ClientConfig.setSmartRouting()`), as this optimization has zero effect.